### PR TITLE
Feature/reactions

### PR DESCRIPTION
**Pull request by Arkitecht**
Adds the ability to "react" to a post.
It will show the quantity if more than 1 of the same reaction exists. 

Clicking a reaction you have already reacted, removes the reaction

Some screenshots:
image
Screen Shot 2022-10-25 at 9 06 31 AM
Screen Shot 2022-10-25 at 9 06 45 AM
Screen Shot 2022-10-25 at 9 06 54 AM

### CODE REVIEW

**Comment by andreaselia** [comment]
Looking good! A few quick suggestions before I get a chance to properly test and look into both you your pull requests would be:

- We should look at not including assets for features that aren't enabled (e.g. the emojis, as it's a 177kb file, so we don't want to include it if people don't want to use the emojis).
- I might update some of the styling myself, but a suggestion would be to include the reactions actually inside the post box so that it's within the padding of the post, instead of on the border.

Thanks again for your contributions, and for your response to the issue in that other thread. Would you be interested in becoming a maintainer for this project? Feel free to reach out to me on Discord if you use it: andreaselia#4499

**Comment by Arkitecht** [comment]
@andreaselia - Thanks for the feedback

1. I can update to store the emojis.js separately, and then only include it if the feature is enabled. 
2. I can update the UI as well, went back and forth on a couple of options.

I would be happy to join as a maintainer. I will hit you up on Discord.

File: generate-emojis-js.php
Comment (andreaselia) [inline generate-emojis-js.php:1]: What is this file? Should it exist, be a command, or maybe be deleted and ship with just a preset of emojis? 🤔

**Review by andreaselia — CHANGES_REQUESTED** [review]
cc: @tomirons

File: routes/api.php
Comment (Arkitecht) [reply routes/api.php:37]: I have updated these

File: routes/web.php
Comment (Arkitecht) [reply routes/web.php:45]: i have updated these

File: src/Policies/PostPolicy.php
Comment (Arkitecht) [reply src/Policies/PostPolicy.php:116]: Class imported

File: generate-emojis-js.php
Comment (Arkitecht) [reply generate-emojis-js.php:1]: This file grabs the current list of emojis from an API and builds the emojis.js file. Its only needed as part of the package so we can occasionally update the emoji list, therefor I don't think it needs to be (or should be) a command, but included it to make updating easy in the future.

File: resources/views/components/post-item.blade.php
Comment (Arkitecht) [reply resources/views/components/post-item.blade.php]: I have removed this

File: src/Services/ReactionService.php
Comment (andreaselia) [reply src/Services/ReactionService.php:41]: I was going to suggest `wasRecentlyCreated` when I first started reading your message, but I do also agree with it having the potential to not being as easy to follow, but then it is also shorter in terms of the code needing to be written, so I'll leave you to have the final call on this one and I'll support whichever choice you go with 😁

File: routes/web.php
Comment (andreaselia) [reply routes/web.php]: @Arkitecht do you think we could also update the name of the "delete" route to "unreact" unless you see any issues in doing so?

**Review by andreaselia — APPROVED** [review]

**Pull request merged**
